contracts/src/LightClient.sol

```diff
@@ -292,4 +322,26 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             )
         );
     }
+
+    /// @notice Update the address of the permissioned prover
+    /// @dev this address is only considered when the `permissionedProverMode` is set to true
+    function updatePermissionedProver(address _prover) public onlyOwner {
+        if (_prover == address(0)) {
+            revert InvalidAddress();
+        }
+        permissionedProver = _prover;
+        emit PermissionedProverUpdated(_prover);
+    }
+
+    /// @notice Set the permissioned prover mode
+    /// @dev when this is set to true, only the `permissionedProver` is able to execute the
+    /// `newFinalizedState` method
+    function setPermissionedProverMode(bool mode) public onlyOwner {
```

Comment:
Is it useful to allow setting the mode to the mode that it's already in? I feel like we don't need to emit an enabled/disabled event in case the mode wasn't changed.

Reply:
yes that's true! addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/c6ff57f7d67653b9bc5bb3bf69bda7f5609f779e